test/e2e/network/network_policy.go

```diff
@@ -55,50 +76,83 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
 		e2eskipper.SkipIfNodeOSDistroIs("windows")
 	})
 
-	ginkgo.Context("NetworkPolicy between server and client", func() {
-		ginkgo.BeforeEach(func() {
-			ginkgo.By("Creating a simple server that serves on port 80 and 81.")
-			podServer, service = createServerPodAndService(f, f.Namespace, "server", []int{80, 81})
+	beforeEach := func(protocol connProtocol) {
+		ginkgo.By("Creating a simple server that serves on two ports")
+		if protocol == tcp {
+			podServer, service = createServerPodAndService(f, f.Namespace, fmt.Sprintf("server-%s", protocol), []int{80, 81})
+		} else {
+			podServer, service = createServerPodAndService(f, f.Namespace, fmt.Sprintf("server-%s", protocol), []int{}, []sctpPort{sctpPort{80, 82}, sctpPort{81, 83}}...)
```

Comment:
so as discussed on slack yesterday, I'd avoid major rewrites to the NetworkPolicy code. Just add one or two new SCTP tests and don't change the existing tests.